### DISCUSSION TITLE
Keycloak 17 support?

### DISCUSSION THREAD

**Discussion by knutwannheden** [quarkusio/quarkus#24029] [Q&A] [answered]
So far we have been using the bare JWT extension (`quarkus-smallrye-jwt`). Now we want to switch to `quarkus-oidc` and the first thing I noticed was considerably longer build times, as the build runs tests which end up starting the OIDC dev-service.

Recently I read that the new Keycloak version 17 is now built on Quarkus, so I was thinking this may help speed up the builds a little again. But looking at the dev-service I notice it is based on `keycloak-x` which already is Quarkus-based. The new Keycloak 17 Docker image is still smaller, so it could of course still be faster.

I was wondering if the new Keycloak 17 images are supported already by the OIDC dev service and if so how it needs to be configured (also with respect to the config property `keycloak-x-image`).

**Comment by sberyozkin** [comment]
You can set `quarkus.devservices.keycloak.image-name=quay.io/keycloak/keycloak-x:17.0.0`. `keycloak-x-image` property is only there to support running custom Keycloak-x images which have no `keycloak-x` in their name

**Reply by knutwannheden** [reply]
@sberyozkin I tested with `quarkus.keycloak.devservices.image-name=quay.io/keycloak/keycloak:17.0.0` (note the difference both in the config key and value) but wasn't able to get it to work. Thus it looks like Keycloak 17 currently isn't supported out-of-the-box.

**Comment by sberyozkin** [comment, marked as answer]
@knutwannheden Can you please add `quarkus.keycloak.devservices.keycloak-x-image=true` ?

**Reply by knutwannheden** [reply]
When I set this additional property the Keycloak dev service starts up correctly. It started in ~12 sec instead of ~14 sec so it does appear to be marginally faster. (I was hoping it would be much faster :-)) I still need to do some testing to see if it all works, but at least I was now able to start the server. Thanks!

**Reply by sberyozkin** [reply]
@knutwannheden Thanks; also see #24033

**Reply by sberyozkin** [reply]
There might be some default value changes for some properties if the KC-X image is now a default when I'll get to updating Quarkus to KC 17
